From 9aa404a57823c9fbf2bf2e0189d31a2b0d8bf3b9 Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@redhat.com>
Date: Thu, 25 Feb 2010 18:41:13 +0530
Subject: [PATCH] pc: Add a Fedora-13 machine type that contains backports from upstream

We have a few features backported from the upstream 0.13 machine type
in the repo here.

Add a 'fedora-13' machine type and use that by default so that users can
fall back to a released upstream machine type if desired.

The fedora-13 machine type has the new multiport-supported virtio-serial
and vhost-net patches as of now not present in the 0.12 machine type.

This is based on upstream commit
d76fa62dba54a156ca0f5e79eb33756c9015e02c

Signed-off-by: Amit Shah <amit.shah@redhat.com>
---
 hw/pc.c |   19 +++++++++++++++++--
 1 files changed, 17 insertions(+), 2 deletions(-)

diff --git a/hw/pc.c b/hw/pc.c
index 56be728..26c65c1 100644
--- a/hw/pc.c
+++ b/hw/pc.c
@@ -1327,11 +1327,9 @@ void cmos_set_s3_resume(void)

 static QEMUMachine pc_machine = {
     .name = "pc-0.12",
-    .alias = "pc",
     .desc = "Standard PC",
     .init = pc_init_pci,
     .max_cpus = 255,
-    .is_default = 1,
 };

 static QEMUMachine pc_machine_v0_11 = {
@@ -1416,3 +1414,20 @@ static void pc_machine_init(void)
 }

 machine_init(pc_machine_init);
+
+/* Fedora machine types */
+static QEMUMachine pc_machine_f13 = {
+    .name = "fedora-13",
+    .alias = "pc",
+    .desc = "Standard PC",
+    .init = pc_init_pci,
+    .max_cpus = 255,
+    .is_default = 1,
+};
+
+static void fedora_machine_init(void)
+{
+    qemu_register_machine(&pc_machine_f13);
+}
+
+machine_init(fedora_machine_init);
-- 
1.6.6.1

